src/utils/variable_manipulations.jl

```diff
@@ -51,6 +53,64 @@ Arguments:
     return ρa == 0 ? ρχ / ρ : weight * ρaχ / ρa + (1 - weight) * ρχ / ρ
 end
 
+"""
+    scalar_names(field)
+
+Filters and returns the names of the variables from a given state
+vector component, excluding `ρ`, `ρe_tot`, and `uₕ`
+
+Arguments:
+
+- `field`: A component of the state vector `Y.c`.
+
+Returns:
+
+- A `Tuple` of `ClimaCore.MatrixFields.FieldName`s corresponding to the tracers.
+"""
+scalar_names(field) =
+    unrolled_filter(MatrixFields.top_level_names(field)) do name
+        !(name in (@name(ρ), @name(ρe_tot), @name(uₕ)))
+    end
+
+"""
+    foreach_scalar(f::F, Yₜ, Y) where {F}
```

Comment:
```suggestion
    foreach_tracerf::F, Yₜ, Y) where {F}
```
and consistently throughout, `scalar` -> `tracer`